### PR TITLE
support creating job in folder

### LINKED ISSUE #65: Cannot created nested job?
```
jenkins.CreateJob("xml..", "app/job/buildjob/"); 
```

returns: /  is an unsafe character

### PR DESCRIPTION
**Pull request by supereagle**
fix #65. 

@bndr  PTAL. If it is ok, I will add the test cases.

### CODE REVIEW

File: jenkins.go [hunk 1]
```diff
@@ -210,6 +210,32 @@ func (j *Jenkins) CreateNode(name string, numExecutors int, description string,
 	return nil, errors.New(strconv.Itoa(resp.StatusCode))
 }
 
+// Create a new folder
+// This folder can be nested in other parent folders
+// Example: jenkins.CreateFolder("newFolder", "grandparentFolder", "parentFolder")
+func (j *Jenkins) CreateFolder(name string, parents ...string) (*Folder, error) {
+	folderObj := &Folder{Jenkins: j, Raw: new(FolderResponse), Base: "/job/" + strings.Join(append(parents, name), "/job/")}
```
Comment (bndr) [inline jenkins.go]: How does the end URL look like if for example I call the function like this: jenkins.CreateFolder("newFolder", "grandparentFolder", "parentFolder"). Should it look like "https://jenkins_url/job/grandparentFolder/job/parentFolder/createItem"?

**Comment by bndr** [comment]
We should definitely add some tests for this feature.

File: jenkins.go (hunk 1 above)
Comment (supereagle) [inline jenkins.go]: Yes.

**Comment by supereagle** [comment]
@bndr  Thanks for your review. I will ping you after the test cases are added.

**Comment by supereagle** [comment]
I am a little busy these days, will add test cases and doc at this weekend.

**Comment by supereagle** [comment]
@bndr Have added the test case, sorry for the delay. But the CI failed when run test cases. Does the test environment need to install the Folders Plugin?

Error log:
```
Goroutine 92 (running) created at:
  github.com/bndr/gojenkins.TestConcurrentRequests()
      /home/travis/gopath/src/github.com/bndr/gojenkins/jenkins_test.go:180 +0x4b
  testing.tRunner()
      /home/travis/.gimme/versions/go/src/testing/testing.go:747 +0x16c
==================
--- FAIL: TestCreateFolder (0.45s)
	Error Trace:	jenkins_test.go:193
	Error:      	Expected nil, but got: &errors.errorString{s:"No item type ?com.cloudbees.hudson.plugins.folder.Folder? is known"}
	Error Trace:	jenkins_test.go:194
	Error:      	Expected value not to be nil.
	testing.go:700: race detected during execution of test
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7c6f31]
```

**Comment by bndr** [comment]
Yeah it's an optional plugin, I'll take a look into it

**Comment by supereagle** [comment]
@bndr  What about the test environment? Do you need any help from me?

**Pull request merged**